node/src/service.rs

```diff
@@ -414,669 +449,86 @@ where
 	Ok((task_manager, client))
 }
 
-// We use this as the RuntimeApi definition in the following function so that we don't need to specify so many generic traits.
-// Using generics would be more flexible, but this is fine for now.
-use foucoco_runtime::RuntimeApi as FoucocoRuntimeApi;
-/// Start a node that exposes the RPC endpoints for the spacewalk pallets for the Foucoco runtime.
-#[sc_tracing::logging::prefix_logs_with("Parachain")]
-async fn start_node_impl_foucoco<Executor>(
-	parachain_config: Configuration,
-	polkadot_config: Configuration,
-	collator_options: CollatorOptions,
-	id: ParaId,
-	hwbench: Option<sc_sysinfo::HwBench>,
-) -> sc_service::error::Result<(
-	TaskManager,
-	Arc<TFullClient<Block, FoucocoRuntimeApi, NativeElseWasmExecutor<Executor>>>,
-)>
+/// Build the import queue for the parachain runtime.
+fn build_import_queue<RuntimeApi, Executor>(
```

Comment:
The diff algorithm gets a little confused here and does not properly align the old and new code. `build_import_queue` occurs below the four `start_node_impl_<...>` functions and is basically unchanged. I could just use the greatly simplified type guard `RuntimeApi::RuntimeApi: ParachainRuntimeApiImpl`.